packages/firebase_auth/firebase_auth_web/lib/firebase_auth_web.dart

```diff
@@ -138,6 +99,104 @@ class FirebaseAuthWeb extends FirebaseAuthPlatform {
     return FirebaseAuthWeb._();
   }
 
+  bool _cancelUserStream = false;
+  bool _cancelIdTokenStream = false;
+
+  void _createStreamListener(String appName, StateListener stateListener) {
+    if (StateListener.authStateChange == stateListener) {
```

Comment:
Maybe using a switch/case here would be easier to read and make future addition more obvious